opencog/atomspace/Handle.cc

```diff
@@ -34,6 +34,8 @@ using namespace opencog;
 const Handle Handle::UNDEFINED(ULONG_MAX);
 
 Handle::Handle(AtomPtr atom) : _uuid(atom->_uuid), _ptr(atom) {}
+Handle::Handle(void* atom_ptr) :  _uuid( (*((AtomPtr*)atom_ptr))->_uuid ),
+                                    _ptr(*((AtomPtr*)atom_ptr)) {}
```

Comment:
The fragment `*((AtomPtr*)atom_ptr)` is the `shared_ptr<Atom>` that will get passed by reference into this constructor:

```
shared_ptr( const shared_ptr& r )
```

for the shared_ptr template class. So the reference count will be incremented when this copy gets made as in normal shared_ptr semantics. This is the first and only copy that gets made, here, in the constructor for the C++ Handle (cHandle in Cython). No other copies are made in the process of object initialization and no pointers to the underlying C++ Atom object are passed around anywhere.